### PR TITLE
align, focus preferred, css fixes

### PR DESCRIPTION
**Pull request by Mipme**
- Adds the align attribute.
- Prefers blur event on focusable elements over mouseleave.
- Sets the z-index on :host
- includes Pull Request 090d221 to patch if offsetParent is not the body

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,3 +2,12 @@
 
 `<paper-tooltip>` is a Material Design label that appears when the user hovers
 over an element with the cursor, or focuses the element with the keyboard.
+
+If the target is focusable, the focus and blur event are preferred over the mouse-events.
+So if the cursor is still in an input field, the tooltip stays althought the mouse was moved elsewhere.
+
+## Attributes
```
Comment (notwaldorf) [inline README.md]: This isn't needed. They should be pulled automatically from comment in the `properties` section.

File: paper-tooltip.html [hunk 2]
```diff
@@ -157,24 +158,26 @@
...
       },
 
       detached: function() {
-        if (this._target) {
-          this.unlisten(this._target, 'mouseenter', 'show');
-          this.unlisten(this._target, 'focus', 'show');
-          this.unlisten(this._target, 'mouseleave', 'hide');
-          this.unlisten(this._target, 'blur', 'hide');
-        }
+        var target = this.target;
```
Comment (notwaldorf) [inline paper-tooltip.html]: Why is this needed? When the tooltip gets re-attached, the `target` is re-computed again anyway.

File: paper-tooltip.html [hunk 3]
```diff
@@ -183,27 +186,32 @@
...
+        if (this.target.focused) return;
+
         this._showing = false;
         this.playAnimation('exit');
       },
 
       _setPosition: function() {
-        if (!this._target)
-          return;
+        var target = this.target;
```
Comment (notwaldorf) [inline paper-tooltip.html]: The idea here is that I don't want to called `this.target` every time the tooltip is show, as this isn't good for performance

File: README.md [hunk 4]
```diff
@@ -2,3 +2,12 @@
...
+
+* `align` - ['left', 'center', 'right']. Aligns the tooltip accordingly
```
Comment (notwaldorf) [inline README.md]: I don't think this is needed either. There's a  `-paper-tooltip` mixin available, where you can do all your styling

File: README.md (hunk 1 above)
Comment (Mipme) [inline README.md]: Ok, but where? There are neither focus nor blur properties in that section 'cause we don't need these as props. I reckon it's a plain helpful description of the general functionality...

File: paper-tooltip.html (hunk 2 above)
Comment (Mipme) [inline paper-tooltip.html]: Good question. It originating from the other pull request i've incorporated. Shall i remove it?

File: README.md (hunk 4 above)
Comment (Mipme) [inline README.md]: yep, target is for, will correct it, is a mistake.

@alignment: the benefit of the computed left/right is, when using a target element which is less wide than the container. A simple CSS align right would put the tooltip to the complete right of the container. This way the tooltip is positioned at the right end of the target.

**Comment by goldieep** [comment]
can you please merge this? i demand this to be merged soon

**Comment by notwaldorf** [comment]
Closing this, since tooltip positioning already landed in a different PR
